### PR TITLE
[UI] Fix misalligned icons

### PR DESCRIPTION
**Pull request by earl-warren**
Fixes: https://codeberg.org/forgejo/forgejo/pulls/823

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Keeping adding "gt-df gt-ac" is not the right approach. If you search `<i class="icon icon-octicon">` , there are a lot.

So, I do not think it's urgent to introduce this patch, I will propose a complete fix for SVG alignment.

**Comment by silverwind** [comment]
> Keeping adding "gt-df gt-ac" is not the right approach. If you search `<i class="icon icon-octicon">` , there are a lot.
> 
> So, I do not think it's urgent to introduce this patch, I will propose a complete fix for SVG alignment.

I'm okay with it. As I said earlier, SVG vertical centering with text has no univeral fix that works for every font and icon size.

File: templates/repo/issue/view_content/update_branch_by_merge.tmpl [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {{if and (gt $.Issue.PullRequest.CommitsBehind 0) (not $.Issue.IsClosed) (not $.Issue.PullRequest.IsChecking) (not $.IsPullFilesConflicted) (not $.IsPullRequestBroken)}}
 	<div class="ui divider"></div>
 	<div class="item item-section">
-		<div class="item-section-left">
-			<i class="icon icon-octicon">{{svg "octicon-alert"}}</i>
-			{{$.locale.Tr "repo.pulls.outdated_with_base_branch"}}
+		<div class="item-section-left gt-df gt-ac">
+			<i class="icon icon-octicon gt-df gt-ac">{{svg "octicon-alert"}}</i>
```
Comment (silverwind) [inline templates/repo/issue/view_content/update_branch_by_merge.tmpl]:
```suggestion
			<i class="icon icon-octicon>{{svg "octicon-alert"}}</i>
```

This is likely unnecessary given as the `<i>` should not have any space inside in which to center, e.g. it wraps the `<svg>` exactly.

**Comment by wxiaoguang** [comment]
> SVG vertical centering with text has no univeral fix that works for every font and icon size.

I think we can introduce a helper CSS class like this:

```
<button class="with-flex-icon">
   <svg ...>
   text ...
   <svg ...>
   text
   <svg ...>
</button>
```

Then let the first "svg" get margin-right, the last "svg" get "margin-left", inside "svg" gets margin-left&margin-right.

And we need to remove all "icon icon-oction" from code, it's a no-op IMO.

**Comment by silverwind** [comment]
> I think we can introduce a helper CSS class like this:

I don't mind the explicit `gt-df gt-ac` so much, they are true to CSS so to say and their meaning is immediately clear, which is not the case with a `with-flex-icon`.

> And we need to remove all "icon icon-oction" from code, it's a no-op IMO.

Agree, `<i>` wrappers are unnecessary, except for the cases where Fomantic has explicit selectors for it, which it does for `i.icon` in some cases.

**Comment by silverwind** [comment]
I do believe this is not a complete fix for alle the messages that can appear in that box:

image

@earl-warren can you give the other messages the same treatment?

**Comment by silverwind** [comment]
I don't agree with introducing magical classes, they will hurt in the long run once requirements get more complex than "a button with an icon", just use the simple CSS helpers in these cases for now please.

**Comment by wxiaoguang** [comment]
Sorry I don't understand the point"magical classes". If such design is called "magical", then every modern framework is full of magical classes. Are they all wrong?

**Comment by silverwind** [comment]
It depends, common classes can help in a few clearly defined use cases. I do see a possibility of having a `text-with-icons` class helper, equivalent to like `gt-df gt-ac gt-fw gt-gap-2`.

**Comment by wxiaoguang** [comment]
What do you think about:  Use flex to alignment SVG and text #25163  ?

**Comment by earl-warren** [comment]
Closing in favor of https://github.com/go-gitea/gitea/pull/25163

**Comment by wxiaoguang** [comment]
> Keeping adding "gt-df gt-ac" is not the right approach. If you search `<i class="icon icon-octicon">` , there are a lot.
> 
> So, I do not think it's urgent to introduce this patch, I will propose a complete fix for SVG alignment.

The complete fix is ready:  Use flex to align SVG and text #25163

**Pull request closed without merging**